### PR TITLE
Guidelines for good commit messages

### PR DESCRIPTION
**Pull request by raoulmillais**
What I am suggesting:

- [x] Gets out of the way;
- [x] Is not restricting;
- [x] Is safe, production ready, and project members have experience with it;
- [x] Promotes good coding practices.

### CODE REVIEW

File: gitiquette/separate-subject-from-body.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+Derezz the master control program
+
+MCP turned out to be evil and had become intent on world domination. This commit
+throws Tron's disc into MCP (causing its deresolution) and turns it back into a
+chess game.
+```
+
+This is not so easy to commit this with the -m switch. You really
+need a proper editor. If you do not already have an editor set up for use with
+git at the command line, read this section of Pro Git.
```
Comment (nfcampos) [inline gitiquette/separate-subject-from-body.md]: what section of Pro Git? I think we can lose this sentence, no?

File: gitiquette/limit-subject-line.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+## Limit the subject line to 50 characters
+
+[Back](../gitiquette.md)
+
+50 characters is not a hard limit, just a rule of thumb. Keeping subject lines at this length ensures that they are readable, and forces the author to think for a moment about the most concise way to explain what's going on.
+
+Tip: If you're having a hard time summarizing, you might be committing too many
+changes at once. Strive for atomic commits (a topic for a separate post).
```
Comment (nfcampos) [inline gitiquette/limit-subject-line.md]: we can probably lose `(a topic for a separate post)`

File: gitiquette/wrap-body.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+Git never wraps text automatically. When you write the body of a commit message,
+you must mind its right margin, and wrap text manually.
+
+The recommendation is to do this at 72 characters, so that git has plenty of
+room to indent text while still keeping everything under 80 characters overall.
+
+A good text editor can help here. It's easy to configure Vim, for example, to
+wrap text at 72 characters when you're writing a git commit. Traditionally,
+however, IDEs have been terrible at providing smart support for text wrapping in
```
Comment (nfcampos) [inline gitiquette/wrap-body.md]: I think we can probably do without the comment about IDEs being terrible

File: gitiquette.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+completely, so our efforts should go to reducing it [as much] as possible.
+Commit messages can do exactly that and as a result, a commit message shows
+whether a developer is a good collaborator. If you haven't given much thought to
+what makes a great git commit message, it may be the case that you haven't spent
+much time using git log and related tools. There is a vicious cycle here:
+because the commit history is unstructured and inconsistent, one doesn't spend
+much time using or taking care of it. And because it doesn't get used or taken
+care of, it remains unstructured and inconsistent.
+
+But a well-cared for log is a beautiful and useful thing. git blame, revert,
```
Comment (nfcampos) [inline gitiquette.md]: Maybe put all git commands inside backticks, like `git blame`, to distinguish from text

**Review by nfcampos — COMMENTED** [review]
This looks great other than the minor nits I pointed out

File: gitiquette/wrap-body.md (hunk 3 above)
Comment (raoulmillais) [reply gitiquette/wrap-body.md]: 💯 Definitely

File: gitiquette.md (hunk 4 above)
Comment (raoulmillais) [reply gitiquette.md]: Good spot I tried to do this but I obviosuly missed some

File: gitiquette/limit-subject-line.md (hunk 2 above)
Comment (raoulmillais) [reply gitiquette/limit-subject-line.md]: :+1:

**Review by tomgco — APPROVED** [review]
Looks great, really comprehensive!

**Pull request merged**